spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioConnection.java

```diff
@@ -620,11 +625,27 @@ public void write(int b) throws IOException {
 
 		@Override
 		public void close() {
-			doClose();
+			if (this.selector != null) {
+				try {
+					this.selector.close();
+				}
+				catch (@SuppressWarnings(UNUSED) Exception e) {
```

Comment:
I don't think it is this method responsibility to suppress an exception.
Let it be done in the `TcpNioConnection.close()` instead!

Reply:
Ok

Reply:
I thought that we agreed in the previous discussion that this `catch` is going to be removed...